core/src/main/kotlin/org/evomaster/core/problem/rest/service/ResourceSampler.kt

```diff
@@ -0,0 +1,278 @@
+        if(withDependency){
+            val num = randomness.nextInt(3, config.maxTestSize)
+            dm.sampleRelatedResources(resourceCalls, num, config.maxTestSize)
+        }else{
+            sampleManyResources(resourceCalls)
+        }
+    }
+
+    private fun selectAResource(randomness: Randomness) : String{
+        return randomness.choose(rm.getResourceCluster().filter { r -> r.value.isAnyAction() }.keys)
+    }
+
+    private fun selectAIndResourceHasNonInd(randomness: Randomness) : String{
+        return randomness.choose(rm.getResourceCluster().filter { r -> r.value.isAnyAction() && !r.value.isIndependent() }.keys)
+    }
+
+    private fun selectAComResource(randomness: Randomness) : String{
+        val skip = rm.getResourceCluster().filter { r -> !r.value.isAnyAction() }.keys
+        val candidates = samplingComResourceCounter.filterNot{ r-> r.key.split(separatorResources).any{ir-> skip.contains(ir)} }
+        if(candidates.isEmpty())
+            throw IllegalStateException("there is no any com-resource available")
+        return randomness.choose(candidates.keys)
+
+    }
+
+    private fun initAdHocInitialIndividuals() {
+        adHocInitialIndividuals.clear()
+
+        rm.createAdHocIndividuals(NoAuth(),adHocInitialIndividuals)
+        authentications.forEach { auth ->
+            rm.createAdHocIndividuals(auth, adHocInitialIndividuals)
+        }
+    }
+
+    private fun initAbstractResources(){
+        rm.initResourceNodes(actionCluster, sqlInsertBuilder)
+
+        rm.getResourceCluster().keys.forEach { k->
+            samplingResourceCounter.getOrPut(k){0}
+        }
+
+        rm.getResourceCluster().filter { !it.value.isIndependent() }.keys.forEach { k ->
+            rm.getResourceCluster().filter { !it.value.isIndependent() }.keys.forEach { k1 ->
+                if(k != k1){
+                    samplingComResourceCounter.getOrPut(k+separatorResources+k1){0}
+                }
+            }
+        }
+    }
+
+    override fun hasSpecialInit(): Boolean {
+        return adHocInitialIndividuals.isNotEmpty() && config.probOfSmartSampling > 0
+    }
+
+    override fun resetSpecialInit() {
+        initAdHocInitialIndividuals()
+    }
+
+    override fun feedback(evi : EvaluatedIndividual<RestIndividual>) {
+        if(config.resourceSampleStrategy.requiredArchive && evi.hasImprovement && evi.individual is RestIndividual)
+            evi.individual.sampleSpec?.let { ssc.reportImprovement(it) }
+    }
+    private fun sampleRandomComResource(resourceCalls: MutableList<RestResourceCalls>){
+        val keys = selectAComResource(randomness)
+        var size = config.maxTestSize
+        var num = 0
+        for (key in keys.split(separatorResources)){
+            rm.sampleCall(key, true, resourceCalls, config.maxTestSize)
+            size -= resourceCalls.last().actions.size
+            num++
+        }
```

Comment:
are `num` and `size` actually used here? it looks like you are only writing their values, isn't it?

Reply:
fixed. It is a bug, and `size` should be used when sampling the next set of actions.